lib/spack/spack/binary_distribution.py

```diff
@@ -2029,7 +2086,7 @@ def extract_tarball(spec, download_result, unsigned=False, force=False, timer=ti
             _delete_staged_downloads(download_result)
             shutil.rmtree(tmpdir)
             raise e
-    else:
+    elif layout_version == 1:
```

Comment:
Do we need an `else` case here for unknown layouts?

Reply:
That already throws earlier